### PR TITLE
mail-client/alot: version bump 0.9

### PR DESCRIPTION
**Pull request by GuillaumeSeren**
Closes: https://bugs.gentoo.org/710610
Package-Manager: Portage-2.3.84, Repoman-2.3.20
Signed-off-by: Guillaume Seren <guillaumeseren@gmail.com>

### CODE REVIEW

**Comment by GuillaumeSeren** [comment]
I have upgraded the branch:
* Add a patch (28a4296c7f556c251d71d9502681980d46d9fa55) already in master fix a important issue.
* Upgrade the live-ebuild.

File: mail-client/alot/alot-0.9.ebuild [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+python_compile_all() {
+	if use doc; then
+		emake -C docs html
+		HTML_DOCS=( docs/build/html/. )
+	fi
+}
+
+src_test() {
+	esetup.py test
+}
```
Comment (juippis) [inline mail-client/alot/alot-0.9.ebuild]: Would this work with a new function in the *distutils-r1.eclass* called *distutils_enable_tests*?

File: mail-client/alot/alot-0.9.ebuild [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	>=dev-python/configobj-4.7.0[${PYTHON_USEDEP}]
+	dev-python/python-magic[${PYTHON_USEDEP}]
+	>=dev-python/urwid-1.3.0[${PYTHON_USEDEP}]
+	>=dev-python/urwidtrees-1.0[${PYTHON_USEDEP}]
+	>=dev-python/twisted-18.4[${PYTHON_USEDEP}]
+	net-mail/mailbase
+	>=net-mail/notmuch-0.27[crypt,python]
+	"
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
```
Comment (juippis) [inline mail-client/alot/alot-0.9.ebuild]: This is handled by distutils-r1.eclass.

File: mail-client/alot/alot-0.9.ebuild [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	doc? ( dev-python/sphinx[${PYTHON_USEDEP}] )
```
Comment (juippis) [inline mail-client/alot/alot-0.9.ebuild]: This too, 
`distutils_enable_sphinx`

File: mail-client/alot/alot-0.9.ebuild [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+SRC_URI="https://github.com/pazz/alot/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="doc test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	>=app-crypt/gpgme-1.9.0[python,${PYTHON_USEDEP}]
```
Comment (juippis) [inline mail-client/alot/alot-0.9.ebuild]: Version is always satisfied in ::gentoo so requirement can be removed.

File: mail-client/alot/alot-0.9.ebuild [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+	>=dev-python/configobj-4.7.0[${PYTHON_USEDEP}]
```
Comment (juippis) [inline mail-client/alot/alot-0.9.ebuild]: Here too.

File: mail-client/alot/alot-0.9.ebuild [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+	dev-python/python-magic[${PYTHON_USEDEP}]
+	>=dev-python/urwid-1.3.0[${PYTHON_USEDEP}]
+	>=dev-python/urwidtrees-1.0[${PYTHON_USEDEP}]
```
Comment (juippis) [inline mail-client/alot/alot-0.9.ebuild]: These too.

File: mail-client/alot/alot-0.9.ebuild [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+	>=dev-python/twisted-18.4[${PYTHON_USEDEP}]
+	net-mail/mailbase
+	>=net-mail/notmuch-0.27[crypt,python]
```
Comment (juippis) [inline mail-client/alot/alot-0.9.ebuild]: And this.

**Comment by GuillaumeSeren** [comment]
Hey @juippis 
I have been working on the cleaning of the ebuild, here the diff of my actual ebuild.

I would like to add the man page, as I noticied that it wasn't in the ebuild,
the commented section works fine without the distutils_enable_sphinx,
but the function did not works with the man page generation (it is commented in the diff).

```diff
diff --git a/mail-client/alot/alot-0.9.ebuild b/mail-client/alot/alot-0.9.ebuild
index 98289e9ec8a..cbc8e866798 100644
--- a/mail-client/alot/alot-0.9.ebuild
+++ b/mail-client/alot/alot-0.9.ebuild
@@ -18,18 +18,16 @@ IUSE="doc test"
 RESTRICT="!test? ( test )"

 RDEPEND="
-       >=app-crypt/gpgme-1.9.0[python,${PYTHON_USEDEP}]
-       >=dev-python/configobj-4.7.0[${PYTHON_USEDEP}]
+       app-crypt/gpgme[python,${PYTHON_USEDEP}]
+       dev-python/configobj[${PYTHON_USEDEP}]
        dev-python/python-magic[${PYTHON_USEDEP}]
-       >=dev-python/urwid-1.3.0[${PYTHON_USEDEP}]
-       >=dev-python/urwidtrees-1.0[${PYTHON_USEDEP}]
+       dev-python/urwid[${PYTHON_USEDEP}]
+       dev-python/urwidtrees[${PYTHON_USEDEP}]
        >=dev-python/twisted-18.4[${PYTHON_USEDEP}]
        net-mail/mailbase
-       >=net-mail/notmuch-0.27[crypt,python]
+       net-mail/notmuch[crypt,python]
        "
 DEPEND="
-       dev-python/setuptools[${PYTHON_USEDEP}]
-       doc? ( dev-python/sphinx[${PYTHON_USEDEP}] )
        test? (
                ${RDEPEND}
                dev-python/mock[${PYTHON_USEDEP}]
@@ -41,18 +39,17 @@ PATCHES=(
        "${FILESDIR}/${PV}-0002-update-reference-to-envelope-body.patch"
        )

-python_compile_all() {
-       if use doc; then
-               emake -C docs html
-               HTML_DOCS=( docs/build/html/. )
-       fi
-}
+distutils_enable_tests setup.py
+distutils_enable_sphinx docs/source

-src_test() {
-       esetup.py test
-}
+# This is new I struggle to get it working along with the distutils_enable_sphinx
+# python_compile_all() {
+#      emake -C docs man
+#      doman docs/build/man/*
+# }

 python_install_all() {
+       use doc && local HTML_DOCS=( docs/source/_build/html/. )
+       # use doc && local HTML_DOCS=( docs/build/htm/. )
        distutils-r1_python_install_all

        insinto /usr/share/alot
```

The error is like sphinx is stuck in man page generation, see:
```sh
 * python3_6: running distutils-r1_run_phase python_compile_all
make -j14 -C docs man
make: Entering directory '/var/tmp/portage/mail-client/alot-0.9/work/alot-0.9/docs'
sphinx-build -b man -d build/doctrees   source build/man
Running Sphinx v2.0.1
making output directory... done
loading intersphinx inventory from http://docs.python.org/objects.inv...
intersphinx inventory has moved: http://docs.python.org/objects.inv -> https://docs.python.org/3/objects.inv
loading intersphinx inventory from http://notmuch.readthedocs.org/en/latest/objects.inv...
intersphinx inventory has moved: http://notmuch.readthedocs.org/en/latest/objects.inv -> https://notmuch.readthedocs.io/en/latest/objects.inv
loading intersphinx inventory from http://urwid.readthedocs.org/en/latest/objects.inv...
intersphinx inventory has moved: http://urwid.readthedocs.org/en/latest/objects.inv -> https://urwid.readthedocs.io/en/latest/objects.inv
building [mo]: targets for 0 po files that are out of date
building [man]: all manpages
updating environment: 35 added, 0 changed, 0 removed
reading sources... [  2%] api/commands
reading sources... [  5%] api/crypto
reading sources... [  8%] api/database
reading sources... [ 11%] api/index
reading sources... [ 14%] api/interface
reading sources... [ 17%] api/overview
reading sources... [ 20%] api/settings
reading sources... [ 22%] api/utils
reading sources... [ 25%] configuration/accounts
reading sources... [ 28%] configuration/config_options
reading sources... [ 31%] configuration/contacts_completion
reading sources... [ 34%] configuration/hooks
reading sources... [ 37%] configuration/index
reading sources... [ 40%] configuration/key_bindings
reading sources... [ 42%] configuration/theming
reading sources... [ 45%] description
reading sources... [ 48%] faq
reading sources... [ 51%] index
reading sources... [ 54%] installation
reading sources... [ 57%] manpage
reading sources... [ 60%] usage/cli_commands
reading sources... [ 62%] usage/cli_options
reading sources... [ 65%] usage/commands
reading sources... [ 68%] usage/crypto
reading sources... [ 71%] usage/first_steps
reading sources... [ 74%] usage/index
reading sources... [ 77%] usage/modes/bufferlist
reading sources... [ 80%] usage/modes/envelope
reading sources... [ 82%] usage/modes/global
reading sources... [ 85%] usage/modes/namedqueries
reading sources... [ 88%] usage/modes/search
reading sources... [ 91%] usage/modes/taglist
reading sources... [ 94%] usage/modes/thread
reading sources... [ 97%] usage/signals
reading sources... [100%] usage/synopsis

looking for now-outdated files... none found
pickling environment... done
checking consistency... done
writing... alot.1 { } done
build succeeded.

The manual pages are in build/man.

Build finished. The manual pages are in build/man.
...
Installing alot script to /var/tmp/portage/mail-client/alot-0.9/image/_python3.6/usr/lib/python-exec/python3.6
 * python3_6: running distutils-r1_run_phase python_install_all
Traceback (most recent call last):
  File "/usr/lib/portage/python3.6/doins.py", line 611, in <module>
    sys.exit(main(sys.argv[1:]))
  File "/usr/lib/portage/python3.6/doins.py", line 602, in main
    os.path.dirname(source)):
  File "/usr/lib/portage/python3.6/doins.py", line 450, in _doins
    return install_runner.install_file(source, os.path.dirname(dest))
  File "/usr/lib/portage/python3.6/doins.py", line 386, in install_file
    return self._ins_runner.run(source, dest_dir)
  File "/usr/lib/portage/python3.6/doins.py", line 195, in run
    sstat = os.stat(source)
FileNotFoundError: [Errno 2] No such file or directory: b'docs/source/_build/html/.'
 * ERROR: mail-client/alot-0.9::local failed (install phase):
 *   dodoc failed
```
It works if I generate both without the helper function like:
```
python_compile_all() {
	emake -C docs man
	doman docs/build/man/*
	use doc && emake -C docs html
}
```

Guillaume

**Comment by GuillaumeSeren** [comment]
I have upgraded only the alot-0.9 for review.

**Comment by GuillaumeSeren** [comment]
Hey @juippis ,
I have upgrade the branch:
* Add live ebuild cleanup
* Add 0.8.1 cleanup
* I use distutils_enable_tests on all versions
* As explained before I am still using the 'emake  -C docs man / html' to be able to generate both html doc and man page.

Guillaume

File: mail-client/alot/alot-0.8.1.ebuild [hunk 8]
```diff
@@ -18,17 +18,16 @@ IUSE="doc test"
 RESTRICT="!test? ( test )"
 
 RDEPEND="
-	>=app-crypt/gpgme-1.9.0[python,${PYTHON_USEDEP}]
-	>=dev-python/configobj-4.7.0[${PYTHON_USEDEP}]
+	app-crypt/gpgme[python,${PYTHON_USEDEP}]
```
Comment (juippis) [inline mail-client/alot/alot-0.8.1.ebuild]: There is no reason to touch this ebuild at all. All these changes require a revbump. Just leave it like it is.

Comment (GuillaumeSeren) [reply mail-client/alot/alot-0.8.1.ebuild]: Ok I keep that change for an other PR.

**Comment by GuillaumeSeren** [comment]
Hey @juippis thank you for your time.

> ```
>  * Messages for package mail-client/alot-0.9:
> 
>  * DISTUTILS_USE_SETUPTOOLS value is probably incorrect
>  *   value:    DISTUTILS_USE_SETUPTOOLS=bdepend (default?)
>  *   expected: DISTUTILS_USE_SETUPTOOLS=rdepend
> ```

I am not sure how do you get this error,
but I added this fix and tested on a chroot clean stage3 (ebuildtester).
```diff
@@ -3,6 +3,7 @@
 
 EAPI=7
 
+DISTUTILS_USE_SETUPTOOLS=rdepend
 PYTHON_COMPAT=( python3_{6,7} )
 
 inherit distutils-r1
```

Guillaume

**Comment by juippis** [comment]
> I am not sure how do you get this error,

https://wiki.gentoo.org/wiki/Package_testing#make.conf
we use these settings to test any ebuild (I use it to test my own ebuilds, too)

**Comment by GuillaumeSeren** [comment]
Thank you @juippis 

> https://wiki.gentoo.org/wiki/Package_testing#make.conf
> we use these settings to test any ebuild (I use it to test my own ebuilds, too)
I will check this, I have been told that ebuildtester (based on docker) is usefull,
I wish we add something built-in to enforce same test everywhere.

Guillaume.

**Pull request closed without merging**